common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -225,76 +246,137 @@ public String encrypt(final String clearText)
 
         final String encryptedText = new String(Base64.encode(blobVerAndEncryptedDataAndIVAndMacDigest,
                 Base64.NO_WRAP), AuthenticationConstants.ENCODING_UTF8);
-        Log.v(TAG, "Finished encryption");
+        Logger.verbose(TAG + methodName, "Finished encryption");
 
         return getEncodeVersionLengthPrefix() + ENCODE_VERSION + encryptedText;
     }
 
     @Override
     public String decrypt(final String encryptedBlob) throws GeneralSecurityException, IOException {
-        final String methodName = "decrypt";
-        Log.v(TAG, "Starting decryption");
+        final String methodName = ":decrypt";
+        Logger.verbose(TAG + methodName, "Starting decryption");
 
         if (StringExtensions.isNullOrBlank(encryptedBlob)) {
             throw new IllegalArgumentException("Input is empty or null");
         }
 
-        int encodeVersionLength = encryptedBlob.charAt(0) - 'a';
-        validateEncodeVersion(encryptedBlob, encodeVersionLength);
+        if (getEncryptionType(encryptedBlob) == EncryptionType.UNENCRYPTED) {
+            Logger.warn(TAG + methodName, "This string is not encrypted. Finished decryption.");
+            return encryptedBlob;
+        }
 
-        final byte[] bytes = Base64.decode(
-                encryptedBlob.substring(1 + encodeVersionLength),
-                Base64.DEFAULT
-        );
+        final String packageName = getPackageName();
+        final List<KeyType> keysForEncryptionType = getKeysForEncryptionType(encryptedBlob, packageName);
+
+        final byte[] bytes = getByteArrayFromEncryptedBlob(encryptedBlob);
+        for (final KeyType keyType : keysForEncryptionType) {
+            try {
+                final SecretKey secretKey = loadSecretKey(keyType);
+                if (secretKey == null) {
+                    continue;
+                }
+
+                String result = decryptWithSecretKey(bytes, secretKey);
+                Logger.verbose(TAG + methodName, "Finished decryption.");
+                return result;
+            } catch (GeneralSecurityException | IOException e) {
+                Logger.error(
+                        TAG + methodName,
```

Comment:
Will we get an error for every value if we have the keysForEncryptionType order wrong as we use StorageHelper? I'm not sure what to recommend.

Reply:
We will see it in the logs, yes.